lib/web/question_router/v1/question_router.ts

```diff
@@ -0,0 +1,1518 @@
+import { QuestionType, Tag } from "@prisma/client"
+import { Decimal } from "@prisma/client/runtime/library"
+import { TRPCError } from "@trpc/server"
+import { z } from "zod"
+import { getBucketedForecasts } from "../../../../pages/api/calibration_graph"
+import { QuestionWithUserAndSharedWith } from "../../../../prisma/additional"
+import {
+  ForecastSchema,
+  QuestionSchema,
+} from "../../../../prisma/generated/zod"
+import {
+  displayForecast,
+  filterToUniqueIds,
+  forecastsAreHidden,
+  getDateYYYYMMDD,
+} from "../../../_utils_common"
+import {
+  backendAnalyticsEvent,
+  updateForecastQuestionMessages,
+} from "../../../_utils_server"
+import { deleteQuestion } from "../../../interactive_handlers/edit_question_modal"
+import { syncToSlackIfNeeded } from "../../../interactive_handlers/postFromWeb"
+import {
+  handleQuestionResolution,
+  undoQuestionOptionResolution,
+  undoQuestionResolution,
+} from "../../../interactive_handlers/resolve"
+import prisma from "../../../prisma"
+import { questionsToCsv } from "../../export"
+import { createNotification } from "../../notifications"
+import { getQuestionUrl } from "../../question_url"
+import { publicProcedure, router } from "../../trpc_base"
+import { assertHasAccess, getQuestionAssertAuthor } from "../assert"
+import { emailNewlySharedWithUsers } from "../email_shared"
+import { getQuestionsUserCreatedOrForecastedOnOrIsSharedWith } from "../get_questions"
+import {
+  getUserByApiKeyOrThrow,
+  getUserFromCtxOrApiKeyOrThrow,
+} from "../get_user"
+import {
+  scrubApiKeyPropertyRecursive,
+  scrubHiddenForecastsAndSensitiveDetailsFromQuestion,
+} from "../scrub"
+import { zodExtraFilters } from "../types"
+
+const exampleQuestionId = "clkqtczp00001l008qcrma6s7"
+
+export const questionRouter = router({
+  getQuestion: publicProcedure
+    .input(
+      z.object({
+        questionId: z
+          .string({
+            description: "ID of the question to fetch",
+          })
+          .optional(),
+        apiKey: z
+          .string({
+            description:
+              "Your Fatebook API key. Get it at fatebook.io/api-setup",
+          })
+          .optional(),
+      }),
+    )
+    .output(QuestionSchema)
```

Comment:
@jonnyspicer This `QuestionSchema` doesn't include `questions.forecasts`, `question.options`,   `sharedWithLists`, `comments`, `tags`, etc. The same is true for `getQuestionsApiProcedure` below, and I think other cases where we use the zod generated schemas as outputs in this file.

This is a moderate issue, because:

1) The schema in the openAPI JSON (as shown on `/api-setup` doesn't include these properties, which are important (e.g., most usecases of getQuestion for MCQs will want to see the options, and in general people will want to see the forecasts on the question). 

2) Additionally, it looks like the Swagger UI only shows parts of the output that conform to the output spec in the web UI when you click execute, so it omits forecasts, options, tags, etc. However, when I test the endpoint using curl I do see the full output.

Do you have a suggestion for how to fix the schema not including prisma-`include`d relations?

I think a temporary fix to allow us to merge this would be to set the schema as `.output(z.any())`, though I agree that it would be nice to have the actual schema shown in the UI setup page without needing to run an example.